analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -63,9 +66,46 @@ public Verifier(VerifierBuilder builder)
             {
                 throw new ArgumentException($"{nameof(builder.Paths)} cannot be empty. Add at least one file using {nameof(builder)}.{nameof(builder.AddPaths)}() or {nameof(builder.AddSnippet)}().");
             }
-            if (builder.Paths.FirstOrDefault(x => !Path.GetExtension(x).Equals(language.FileExtension, StringComparison.OrdinalIgnoreCase)) is { } unexpectedPath)
+            foreach (var path in builder.Paths)
             {
-                throw new ArgumentException($"Path '{unexpectedPath}' doesn't match {language.LanguageName} file extension '{language.FileExtension}'.");
+                ValidateExtension(path);
+            }
+            if (builder.CodeFix is not null)
```

Comment:
Could you please extract the next part to a method? The constructor is way too big.